bindsnet/datasets/davis.py

```diff
@@ -0,0 +1,254 @@
+
+        if subset not in self.SUBSET_OPTIONS:
+            raise ValueError(f"Subset should be in {self.SUBSET_OPTIONS}")
+        if task not in self.TASKS:
+            raise ValueError(f"The only tasks that are supported are {self.TASKS}")
+        if resolution not in self.RESOLUTION_OPTIONS:
+            raise ValueError(
+                f"You may only choose one of these resolutions: {self.RESOLUTION_OPTIONS}"
+            )
+
+        self.task = task
+        self.subset = subset
+        self.resolution = resolution
+
+        # Sets a tag for naming the folder containing the dataset
+        self.tag = ""
+        if self.task == "unsupervised":
+            self.tag += "Unsupervised-"
+        if self.subset == "train" or self.subset == "val":
+            self.tag += "trainval"
+        else:
+            self.tag += self.subset
+        self.tag += "-" + self.resolution
+
+        # Make a unique path for a given instance of davis
+        self.root = os.path.join(root, self.tag)
+        self.download = download
+        self.num_samples = num_samples
+        self.zip_path = os.path.join(self.root, "repo.zip")
+        self.img_path = os.path.join(self.root, "JPEGImages", resolution)
+        annotations_folder = (
+            "Annotations" if task == "semi-supervised" else "Annotations_unsupervised"
+        )
+        self.mask_path = os.path.join(self.root, annotations_folder, resolution)
+        year = (
+            "2019"
+            if task == "unsupervised"
+            and (subset == "test-dev" or subset == "test-challenge")
+            else "2017"
+        )
+        self.imagesets_path = os.path.join(self.root, "ImageSets", year)
+
+        # Check if Davis is installed and download it if necessary
+        self._check_directories()
+
+        # Sets seqence_names to the relevant sequences
+        if sequences == "all":
+            with open(
+                os.path.join(self.imagesets_path, f"{self.subset}.txt"), "r"
+            ) as f:
+                tmp = f.readlines()
+            sequences_names = [x.strip() for x in tmp]
+        else:
+            sequences_names = sequences if isinstance(sequences, list) else [sequences]
+        self.sequences = defaultdict(dict)
+
+        # Sets the images and masks for each sequence
+        for seq in sequences_names:
+            images = np.sort(glob(os.path.join(self.img_path, seq, "*.jpg"))).tolist()
+            if len(images) == 0 and not codalab:
+                raise FileNotFoundError(f"Images for sequence {seq} not found.")
+            self.sequences[seq]["images"] = images
+            masks = np.sort(glob(os.path.join(self.mask_path, seq, "*.png"))).tolist()
+            masks.extend([-1] * (len(images) - len(masks)))
+            self.sequences[seq]["masks"] = masks
+
+        # Creates an enumeration for the sequences for __getitem__
+        self.enum_sequences = self.sequences.items()
+
+    # Returns the number of sequences in the dataset
```

Comment:
Put these comments in doc strings

Reply:
do you want me to change all comments to doc strings? Is that our standard?

Reply:
No, but comments above methods are discouraged. See [this SO post](https://stackoverflow.com/questions/2357230/what-is-the-proper-way-to-comment-functions-in-python). And, and public methods (ones that don't begin with `_`) should have a doc-string.

Reply:
See also [this PEP](https://www.python.org/dev/peps/pep-0257/).

Reply:
ah gotcha, makes sense, I'm on it